src/overlays/actors/ovl_Boss_Tw/z_boss_tw.c

```diff
@@ -156,183 +187,5337 @@ static ColliderCylinderInit D_8094A840 = {
+                             this->actor.scale.z * scaleFactor, MTXMODE_APPLY);
+                func_800D1FD4(&globalCtx->mf_11DA0);
+                gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_boss_tw.c", 8865),
+                          G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                gSPDisplayList(POLY_XLU_DISP++, SEGMENTED_TO_VIRTUAL(D_0601A430));
+            }
+            break;
+
+        case TW_FIRE_BLAST_GROUND:
+            break;
+
+        case TW_ICE_BLAST:
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 195, 225, 235, (s8)this->workf[TAIL_ALPHA]);
+            gSPDisplayList(POLY_XLU_DISP++, SEGMENTED_TO_VIRTUAL(D_0601A998));
+            for (i = 9; i >= 0; i--) {
+                gSPSegment(POLY_XLU_DISP++, 8,
+                           Gfx_TwoTexScroll(
+                               globalCtx->state.gfxCtx, 0, ((this->work[CS_TIMER_1] * 3) + (i * 0xA)) & 0x7F,
+                               (u8)((-this->work[CS_TIMER_1] * 0xF) + (i * 50)), 0x20, 0x40, 1, 0, 0, 0x20, 0x20));
+                tailIdx = ((this->work[TAIL_IDX] - i) + 30) % 30;
+                Matrix_Translate(this->blastTailPos[tailIdx].x, this->blastTailPos[tailIdx].y,
+                                 this->blastTailPos[tailIdx].z, MTXMODE_NEW);
+                scaleFactor = 1.0f - (i * 0.09f);
+                Matrix_Scale(this->actor.scale.x * scaleFactor, this->actor.scale.y * scaleFactor,
+                             this->actor.scale.z * scaleFactor, MTXMODE_APPLY);
+                func_800D1FD4(&globalCtx->mf_11DA0);
+                gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_boss_tw.c", 9004),
+                          G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                gSPDisplayList(POLY_XLU_DISP++, SEGMENTED_TO_VIRTUAL(D_0601AB00));
+            }
+            break;
+
+        case TW_ICE_BLAST_GROUND:
+            break;
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_boss_tw.c", 9013);
+}
+
+void BossTw_DrawDeathBall(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    BossTw* this = THIS;
+    f32 scaleFactor;
+    s16 tailIdx;
+    s16 i;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_boss_tw.c", 9028);
+
+    func_80093D84(globalCtx->state.gfxCtx);
+
+    if (this->actor.params == TW_DEATHBALL_KOUME) {
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 200, 20, 0, (s8)this->workf[TAIL_ALPHA]);
+        gDPSetEnvColor(POLY_XLU_DISP++, 255, 215, 255, 128);
+
+        for (i = 9; i >= 0; i--) {
+            gSPSegment(POLY_XLU_DISP++, 8,
+                       Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, (((this->work[CS_TIMER_1] * 3) + (i * 0xA))) & 0x7F,
+                                        (u8)((-this->work[CS_TIMER_1] * 0xF) + (i * 50)), 0x20, 0x40, 1, 0, 0, 0x20,
+                                        0x20));
```

Comment:
```suggestion
                       Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, (((this->work[CS_TIMER_1] * 3) + (i * 10))) & 0x7F,
                                        (u8)((-this->work[CS_TIMER_1] * 25) + (i * 50)), 0x20, 0x40, 1, 0, 0, 0x20,
                                        0x20));
```
Does `% 128` work here instead of `& 0x7F`? Similarly, can `% 256` replace the `u8` cast?

Reply:
not gonna bother with texscroll tbh, cause these are gonna have to be mangled into macros eventually anyway